File: Hard/1359. Count All Valid Pickup and Delivery Options/1359. Count All Valid Pickup and Delivery Options.py
class Solution:
    def countOrders(self, n: int) -> int:
        MOD = 10**9 + 7
        dp = [0] * (n + 1)
        dp[1] = 1

        for i in range(2, n + 1):
            dp[i] = dp[i - 1] * (2 * i - 1) * i % MOD

        return dp[n]

File: Hard/1359. Count All Valid Pickup and Delivery Options/1359. Count All Valid Pickup and Delivery Options.java
class Solution {
    public int countOrders(int n) {
        long MOD = 1000000007;
        long[] dp = new long[n + 1];
        dp[1] = 1;

        for (int i = 2; i <= n; i++) {
            dp[i] = dp[i - 1] * (2 * i - 1) * i;
            dp[i] %= MOD;
        }

        return (int) dp[n];
    }
}
